### PR TITLE
Homework5: Mikalai Hutsalenka

### PR DESCRIPTION
**Pull request by vorongn**

### CODE REVIEW

File: src/dec_logger.txt [hunk 1]
```diff
@@ -0,0 +1,90 @@
+2020-02-06 16:05:27.005450 / �������: generator / ���������: ['1a', '2a', '3a', '4a']
```
Comment (belovmd) [inline src/dec_logger.txt]: troubles with encoding ^)

File: src/hw05_01_runner.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+"""
+
+
+import mymodule
+
+
+def runner(*args):
+    if not args:
+        args = [elements for elements in
```
Comment (belovmd) [inline src/hw05_01_runner.py]: we should also check if it's callable

File: src/hw05_02.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+from datetime import datetime
+import mymodule
+
+
+def log_dec(function):
+    def wraper(*args, **kwargs):
+        func_result = function(*args, **kwargs)
+        with open('dec_logger.txt', 'a') as my_file:
+            my_file.write('{} / '.format(datetime.now()))
```
Comment (belovmd) [inline src/hw05_02.py:17]: datetime - great!

File: src/hw05_04.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ filename - название файла
+ line_list - список всех значений
+ kolum - номер колонки в котой нужная инфа
+ -----------------------------------------
+ inner_list - список всех значений
+ dict_graph - dict{елемент : колличество}
+"""
+
+
+def reiting(filename, line_list, kolum):
```
Comment (belovmd) [inline src/hw05_04.py]: rating - misprint

File: src/hw05_06.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+"""
+Вводится число найти его максимальный делитель,
+являющийся степенью двойки. 10(2) 16(16), 12(4)
+"""
+
+
+def divider(input_number):
+    degree = 0
+    while input_number % (1 << degree) == 0:
```
Comment (belovmd) [inline src/hw05_06.py:9]: it seems it can be checked with &

**Review by belovmd — CHANGES_REQUESTED** [review]
Please address next issues:

**Review by belovmd — COMMENTED** [review]
please address previous review issues

**Comment by vorongn** [comment]
hw05_01_runner.py - implemented a function check for existence
Hw05_02.py - set default encoding for UTF-8 recording
hw05_04.py - fixed typo rating

File: src/dec_logger.txt (hunk 1 above)
Comment (vorongn) [reply src/dec_logger.txt]: fix

File: src/hw05_01_runner.py (hunk 2 above)
Comment (vorongn) [reply src/hw05_01_runner.py]: implemented a function check for existence

File: src/hw05_04.py (hunk 4 above)
Comment (vorongn) [reply src/hw05_04.py]: fixed typo rating

File: src/hw05_01_runner.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+def runner(*args):
+    all_func = [elements for elements
+                in dir(mymodule) if elements[:2] != '__']
+    if not args:
+        args = all_func
+    for attr in args:
+        if attr in all_func:
+            run_func = getattr(mymodule, attr)
+            run_func()
```
Comment (belovmd) [inline src/hw05_01_runner.py:22]: it will fail attr not a function

File: src/hw05_06.py (hunk 5 above)
Comment (belovmd) [reply src/hw05_06.py:9]: it will work with &

**Review by belovmd — COMMENTED** [review]
8

**Pull request closed without merging**